pkg/compactor/compactor_test.go

```diff
@@ -203,14 +206,13 @@ func TestCompactor_ShouldIterateOverUsersAndRunCompaction(t *testing.T) {
 	`)))
 }
 
-func prepare(t *testing.T) (*Compactor, *cortex_tsdb.BucketClientMock, *tsdbCompactorMock, *bytes.Buffer, prometheus.Gatherer) {
+func prepare(t *testing.T, bucketClient *cortex_tsdb.BucketClientMock) (*Compactor, *tsdbCompactorMock, *bytes.Buffer, prometheus.Gatherer) {
 	compactorCfg := Config{}
 	storageCfg := cortex_tsdb.Config{}
 	flagext.DefaultValues(&compactorCfg, &storageCfg)
 	compactorCfg.retryMinBackoff = 0
 	compactorCfg.retryMaxBackoff = 0
 
-	bucketClient := &cortex_tsdb.BucketClientMock{}
```

Comment:
Changes look good to me but I am trying to understand how this was causing tests to fail intermittently since it was still created before calling `newCompactor` down below. Anything I am missing to see here?

Reply:
It was created before `newCompactor()` but mocked later (after the `prepare()` exits).

Reply:
Right, my bad. Changes look more relevant for [this block](https://github.com/cortexproject/cortex/pull/2018/files#diff-e6297dff640b47246ada70cd6663f9c6R152-R157)

Thanks!